### PR TITLE
Load all .glyphspackage glyphs

### PR DESCRIPTION
**Pull request by simoncozens**
In https://github.com/googlefonts/glyphsLib/issues/846#issuecomment-1478023500, Georg said

> But was in really intended to load the .glif file based on the name? I thought that you should read all files and map them based off the name in the file?

This PR does that: loads all files in the `glyphspackage` directory and then reorders them based on the order in `order.plist`.

### CODE REVIEW

File: Lib/glyphsLib/parser.py [hunk 1]
```diff
@@ -101,17 +101,25 @@ def _parse_dict_into_object(self, res, d):
...
-            openstep_plist.load(open(glyphfile, "r"), use_numbers=True)
+    for glyphfile in (package / "glyphs").glob("*.glyph"):
+        glyph = openstep_plist.load(open(glyphfile, "r"), use_numbers=True)
+        data["glyphs"].append(glyph)
+    # Sort according to glyphorder
+    data["glyphs"] = list(
+        sorted(
+            data["glyphs"],
+            key=lambda x: (
+                x["glyphname"] not in glyphorder,  # unknown glyphs at end
```
Comment (anthrotype) [inline Lib/glyphsLib/parser.py:118]: should the unknown glyphs at the end in turn be sorted alphabetically by their string value, to make things deterministic? I am not sure in what order they will be listed by glob.

File: Lib/glyphsLib/parser.py [hunk 2]
```diff
@@ -101,17 +101,25 @@ def _parse_dict_into_object(self, res, d):
 
 
 def load_glyphspackage(package_dir):
-    infofile = os.path.join(package_dir, "fontinfo.plist")
-    orderfile = os.path.join(package_dir, "order.plist")
+    package = Path(package_dir)
+    infofile = package / "fontinfo.plist"
+    orderfile = package / "order.plist"
```
Comment (anthrotype) [inline Lib/glyphsLib/parser.py:106]: if in the end the order.plist is (unlike the UFO contents.plist) just way to sort glyphs but not an actual index that fully determines what's in the font, can it be omitted altogether?

Comment (simoncozens) [reply Lib/glyphsLib/parser.py:106]: Keeping the glyphs in their expected order is important though, isn't it?

Comment (anthrotype) [reply Lib/glyphsLib/parser.py:106]: well, not as important as keeping it at all. Just curious, does it actually load in Glyphs.app if you omit one?

Comment (anthrotype) [reply Lib/glyphsLib/parser.py:106]: and then, there also the `glyphOrder` custom parameter that one can set on the GSFont that basically accomplishes the same thing..

Comment (anthrotype) [reply Lib/glyphsLib/parser.py:106]: but I think our current code expects the order.plist to be present...
we unconditionally do:

```
glyphorder = openstep_plist.load(open(orderfile, "r", encoding="utf-8"))
```

File: Lib/glyphsLib/parser.py (hunk 1 above)
Comment (simoncozens) [reply Lib/glyphsLib/parser.py]: I fixed this and simplified that horrible lambda.

File: Lib/glyphsLib/parser.py (hunk 2 above)
Comment (anthrotype) [reply Lib/glyphsLib/parser.py:106]: by the way, it's good practice when opening a file to close it afterwards, here and below we just open. Ideally we do `with open(...):`

Comment (anthrotype) [reply Lib/glyphsLib/parser.py:106]: you said

> If there is no order.plist, then Glyphs.app will still load the file

so I'd expect that we match Glyphs.app behavior, whatever their spec says

**Review by anthrotype — APPROVED** [review]
LGTM, thanks!

**Pull request merged**